Shared/Legacy/LegacyMapper.swift

```diff
@@ -0,0 +1,230 @@
+//
+//  LegacyMapper.swift
+//  RelistenShared
+//
+//  Created by Jacob Farkas on 8/12/18.
+//  Copyright © 2018 Alec Gorge. All rights reserved.
+//
+
+import Foundation
+
+class LegacyMapper {
+    public struct GenericImportError : Error { }
+    
+    private let api : RelistenLegacyAPI = RelistenLegacyAPI()
+    private let callbackQueue : DispatchQueue = DispatchQueue(label: "net.relisten.importer")
+    
+    // (farkas) Holy crap this is a mess. I'm hoping to replace this with a state machine soon. If you're reading this I failed at doing that :(
```

Comment:
I'm going to replace this with an NSOperation-based state machine next, but I won't get to that until after Curveball and I figured checking this in now was better than nothing.